### ISSUE TITLE
Histograms in Table 1

### ISSUE THREAD

**Issue by tompollard** [tompollard/tableone#132]
Not sure how easy this would be to implement, but making a note of this idea: https://twitter.com/DocEd/status/1572603047590916097

> why don’t we just have a little histogram for each variable in table 1, in additional to the standard descriptors?

**Comment by jraffa** [comment]
You mean like this?


![image](https://user-images.githubusercontent.com/8632753/191588376-b4dbb48a-1959-44fe-9fb7-a7f683f4fcfc.png)

**Comment by tompollard** [comment]
Yes! I assume this is https://cran.r-project.org/web/packages/skimr/vignettes/skimr.html? 

Looks like it might be possible to do this with Pandas DataFrames, e.g.: https://stackoverflow.com/questions/53468558/adding-image-to-pandas-dataframe

**Comment by jraffa** [comment]
I think the histograms are represented as text:
▇▅▃▅▇

**Comment by tompollard** [comment]
Yeah, it looks like these would require each column to be mapped to an 1/8th: https://www.compart.com/en/unicode/block/U+2580. Seems doable...

**Comment by MartinBernstorff** [comment]
Just want to let you guys know that `skimpy` has this implemented, so it should be real easy to learn from 👍 

https://github.com/aeturrell/skimpy/blob/9a96f252fca01780425bbbc6d6c62165ce9ccf20/src/skimpy/__init__.py#L273

**Comment by MartinBernstorff** [comment]
Glad to hear it! I know this isn't on topic, but just want to say I spent 1,5 days implementing something like `tableone,` found your solution, and deleted all my code. Super happy to find a good, mature solution 👍

**Comment by tompollard** [comment]
@MartinBernstorff thanks for your kind words and I'm glad to hear it has been helpful! We've neglected the package recently so I'm hoping to spend a little time soon making some much needed updates, fixes, etc.

**Comment by tompollard** [comment]
I'm not clear what to do in the case where a groupby variable is specified. Do we: (1) plot a distribution for each separate group, i.e. if there are 4 groups, add 4 columns each with a separate histogram; or (2) plot a single histogram of the overall population? Any preference @DocEd?

**Comment by DocEd** [comment]
Tom, i'm delighted that you saw this and picked it up! I imagine an extra histogram column per group, and if there is space, an overall histogram column at the end. a little like histospark works in R (i think a hadley wickham package).

Or if space was at a premium, the summary numbers could be given as normal, and a soft background image of the histogram behind the text within the table cell could be used.

**Comment by DocEd** [comment]
just seen the rest of this thread. yep, exactly like above.

**Comment by tompollard** [comment]
Thanks everyone! `show_histograms` added in v0.9.5: https://pypi.org/project/tableone/


```
╭───────────────────┬──────┬───────────┬──────────────┬──────────────┬───────────────┬───────────────┬─────────────────────╮
│                   │      │ Missing   │ 0            │ 1            │ 0 Histogram   │ 1 Histogram   │ Overall Histogram   │
├───────────────────┼──────┼───────────┼──────────────┼──────────────┼───────────────┼───────────────┼─────────────────────┤
│ n                 │      │           │ 864          │ 136          │               │               │                     │
│ Age, mean (SD)    │      │ 0         │ 64.0 (17.4)  │ 71.7 (14.0)  │ ▂▂▄▄▆▇█▇      │ ▁▂▃▄▄▆██      │ ▂▂▃▄▆▇█▇            │
│ SysABP, mean (SD) │      │ 291       │ 115.4 (38.3) │ 107.6 (49.4) │ ▂▁▁▂█▇▄▁      │ ▄▁▂▆██▄▂      │ ▂▁▁▃█▆▃▁            │
│ Height, mean (SD) │      │ 475       │ 170.3 (23.2) │ 168.5 (11.3) │ ▁▂▆▇▇█▂▁      │ ▂▆▄▅█▄▅▆      │ ▁▃▅▇▇█▂▁            │
│ Weight, mean (SD) │      │ 302       │ 83.0 (23.6)  │ 82.3 (25.4)  │ ▂▆█▆▄▂▁▁      │ ▅▇█▅▃▄▁▂      │ ▃▆█▆▄▃▁▁            │
│ ICU, n (%)        │ CCU  │           │ 137 (15.9)   │ 25 (18.4)    │               │               │                     │
│                   │ CSRU │           │ 194 (22.5)   │ 8 (5.9)      │               │               │                     │
│                   │ MICU │           │ 318 (36.8)   │ 62 (45.6)    │               │               │                     │
│                   │ SICU │           │ 215 (24.9)   │ 41 (30.1)    │               │               │                     │
╰───────────────────┴──────┴───────────┴──────────────┴──────────────┴───────────────┴───────────────┴─────────────────────╯
```
